src/analyze/analyze-chid.c

```diff
@@ -223,6 +252,65 @@ static int smbios_fields_acquire(char16_t *fields[static _CHID_SMBIOS_FIELDS_MAX
         return 0;
 }
 
+static int edid_parse(sd_device *drm_card, char16_t **ret_panel) {
+        assert(drm_card);
+        assert(ret_panel);
+        int r;
+        const char *edid_content = NULL;
+        size_t edid_size = 0;
+        r = sd_device_get_sysattr_value_full(drm_card, "edid", &edid_content, &edid_size);
+        if (r < 0)
+                return r;
+
+        EdidHeader header;
+        if (!edid_parse_blob(edid_content, edid_size, &header))
+                return -EINVAL;
```

Comment:
while there is no strict rule we typically return EINVAL for invalid invocations, but not for externally provided invalid data, where we'd rather return EBADMSG or so, maybe do that here too